component/client-handler/src/test/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/validator/JWTValidatorTest.java

```diff
@@ -215,41 +215,54 @@ public Object[][] createJWT() throws Exception {
+                {jsonWebToken4, properties3, false, "Request with non existing SP client-id should fail.", null, false},
                 {jsonWebToken5, properties5, true, "JWT replay with preventTokenReuse disabled but " +
-                        "not-expired is not failed"},
-                {jsonWebToken6, properties2, true, "Valid JWT token with custom issuer validation should pass."},
+                        "not-expired is not failed", null, false},
+                {jsonWebToken6, properties2, true, "Valid JWT token with custom issuer validation should pass.", null, false},
                 {jsonWebToken7, properties3, false, "JWT persisted in database with preventTokenReuse " +
-                        "enabled is not failed."},
+                        "enabled is not failed.", null, false},
                 {jsonWebToken9, properties1, false, "JWT persisted in database with preventTokenReuse " +
-                        "disabled is not failed."},
+                        "disabled is not failed.", null, false},
                 {jsonWebToken10, properties4, true, "Valid JWT token with custom audience validation should pass" +
-                        "."},
-                {jsonWebToken11, properties1, false, ""},
-                {jsonWebToken12, properties5, true, ""},
-                {jsonWebToken12, properties5, true, ""},
-                {jsonWebToken13, properties1, false, ""},
-                {jsonWebToken15, properties1, false, ""},
-                {jsonWebToken16, properties4, false, ""},
-                {jsonWebToken17, properties6, false, ""},
-                {jsonWebToken18, properties7, false, ""},
-                {jsonWebToken19, properties1, true, "Unable to use same JTI across tenants."},
+                        ".", null, false},
+                {jsonWebToken11, properties1, false, "", null, false},
+                {jsonWebToken12, properties5, true, "", null, false},
+                {jsonWebToken12, properties5, true, "", null, false},
+                {jsonWebToken13, properties1, false, "", null, false},
+                {jsonWebToken15, properties1, false, "", null, false},
+                {jsonWebToken16, properties4, false, "", null, false},
+                {jsonWebToken17, properties6, false, "", null, false},
+                {jsonWebToken18, properties7, false, "", null, false},
+                {jsonWebToken19, properties1, true, "Unable to use same JTI across tenants.", null, false},
                 {jsonWebToken20, properties1, false, "Duplicated JTI was used in same tenant with " +
-                        "preventTokenReuse enabled."},
+                        "preventTokenReuse enabled.", null, false},
+                {jsonWebToken21, properties1, true, "JWT with registered signing algorithm should pass.", ALG_PS256, true},
+                {jsonWebToken22, properties1, false, "JWT with unregistered signing algorithm should fail.", "RSA265", true}
```

Comment:
need to have a test case where we have signing algs configured in a SP where the SP is not fapi compliant

Reply:
Fixed in e9d0e7eeb2260413659c89164eed65584a7d4278